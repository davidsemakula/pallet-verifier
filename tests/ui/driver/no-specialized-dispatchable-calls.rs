use std::marker::PhantomData;

pub trait Config {}

pub enum Call<T: Config> {
    __Ignore(PhantomData<T>),
    do_something { val: u8 },
}

pub struct Pallet<T: Config>(PhantomData<T>);

pub enum OriginFor<T> {
    Simple,
    Complex(T),
}

impl<T: Config> Pallet<T> {
    pub fn do_something(
        origin: OriginFor<T>,
        val: u8,
    ) {
        // do something
    }
}