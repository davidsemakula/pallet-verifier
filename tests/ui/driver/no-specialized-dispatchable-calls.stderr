error: Failed to generate tractable entry points for any dispatchable function in public interface
   |
   = help: Add some unit tests for dispatchable functions.

error: aborting due to 1 previous error

